superset-frontend/src/components/ListView/utils.ts

```diff
@@ -79,8 +90,34 @@ function mergeCreateFilterValues(list: Filter[], updateObj: any) {
 // convert filters from UI objects to data objects
 export function convertFilters(fts: InternalFilter[]): FilterValue[] {
   return fts
-    .filter(f => typeof f.value !== 'undefined')
-    .map(({ value, operator, id }) => ({ value, operator, id }));
+    .filter(f => {
+      if (typeof f.value === 'undefined') return false;
+      if (Array.isArray(f.value) && !f.value.length) return false;
+      return true;
```

Comment:
how about 
```
.filter(f => !((typeof f.value === 'undefined') || (Array.isArray(f.value) && !f.value.length)))
```

Reply:
Do you actually find that more readable? These are two separate conditions for filtering out a value, it makes sense to keep them seperate imo

Reply:
I do, since you're trying to see if it has a value on each type, but it's just preference. It mainly feels redundant to return a boolean for a boolean conditional. 
How about something like this for readability? 
```
let hasValue = true;
hasValue = !(typeof f.value === 'undefined');
hasValue = !(Array.isArray(f.value) && !f.value.length);
return hasValue;
```
if that doesn't feel like an improvement, we can go with what you have. It's not anything major.

Reply:
ok, updated. I have less of an issue with it now after the prettier formatting.